### ISSUE TITLE
Project Libraries not shown

### ISSUE THREAD

**Issue by JoeBruner** [go-lang-plugin-org/go-lang-idea-plugin#1671]
Not sure if this is actually a bug, but it seems like it should be.  I modified my `Go Libraries` to have no Global libraries, unchecked `Use GOPATH that's defined in system environment` and added two paths under `Project Libraries`.  When looking at the Project view and structure view, I only see 

![image](https://cloud.githubusercontent.com/assets/5578559/7939047/f2a92542-0916-11e5-84be-67385e978658.png)

and 

![image](https://cloud.githubusercontent.com/assets/5578559/7939052/0248c8ea-0917-11e5-9c15-f0f462d6306d.png)

It seems that in addition to the <Go 1.4> and GOPATH (which is there if I don't uncheck the box) that I should be able to see whatever project libraries that I am including.  I know they're there because I can't run my unit tests unless they are included. I don't really think that it should fall under "Well they aren't external because they are project libraries" because the path could be anywhere outside of the project heirachy.  Also, the Structure/Libraries view doesn't imply that they have to be external.

**Comment by dlsniper** [comment]
Hi, sorry unfortunately I haven't fully understood what are you trying to achieve. You want to see which libraries is your project using that are not part of the project? Based on your answer I'll be able to continue the response. Thank you.

**Comment by JoeBruner** [comment]
Correct.  So here's the layout of my System if it helps clarify the situation:

    /usr/local/go  (GO)
    /home/dev/go (Where GOPATH is set to for system wide projects)
    /home/dev/work/work_go_dependencies
    /home/dev/work/bin
    /home/dev/work/src
    /home/dev/work/pkg

`work_go_dependencies` is kind of like a poor man's `godep` since `godep` expects a particular path layout and unfortunately our project can't conform to that.  When working on my work project I don't use the global GOPATH as it has libraries that don't necessarily match the versions that are in `/home/dev/work/work_go_dependencies`.  Instead, I uncheck the box and I set the `module libraries`  to point to `/home/dev/work/work_go_dependencies`.  

I think my confusion arises from the fact that before there was a checkbox to unset the GOPATH, I used to have to set up my GOPATH explicitly in the CLI before starting IDEA, and then I would see the projects under `/home/dev/work/work_go_dependencies` show up under `External Libararies/GOPATH`.   Now that I can unset the `GOPATH` in the IDEA, I've switched to using the project libraries, but I no longer see these external libraries listed under External Libraries.  Ultimately with my setup it doesn't matter because I open the project at the /home/dev/work and clicking on `Scroll from source` will take me to that file.  But that's assuming that deps just happened to be in the hierarchy.  What would happen if instead my external dependencies were in `/usr/local/workdeps` ?

It's kind of a minor annoyance.  It was nice that I could easily see how many external libraries I was pulling into the project, but it's not like I'm going to go in and actively modify those files from my project.

**Comment by zolotov** [comment]
>  I open the project at the /home/dev/work

That's the main thing. External libraries aren't added if they're are included in project itself. They will take a part in go execution/resolving/completion etc. but they won't shown in `project tree/external libraries` because it's part of the project and not external libraries (also in order to avoid some undefined behaviour in project tree UI).

**Comment by JoeBruner** [comment]
Yah, it's just a little odd that the behavior is different based on the location of the files.  In both cases they are "External" libraries, but if they just happen to fall under the current tree then I lose the ability to easily detect that.

**Comment by zolotov** [comment]
I think that in your case they are not external at all. They are editable parts of your project. As soon as you open project in `/home/dev/work/subdireectory` they will be added as external because they're not the part of project but they are reachable by compiler and we have to do something to make them participate in resolution algorithm. Actually it's a little hack of our project model and it's not the thing that should bother users.

**Comment by JoeBruner** [comment]
agreed
